src/Auth/BaseAuthenticate.php

```diff
@@ -146,6 +146,9 @@ protected function _query($username)
         if (!empty($config['contain'])) {
             $options['contain'] = $config['contain'];
         }
+        if (empty($config['username'])) {
```

Comment:
I check for the config value to avoid conflicts (and prevent backward compatibility issues) if someone already uses a config like
```
'finder' => ['auth' => ['username' => 'something']]
```